File: BinarytoDecimal.py
#!/usr/bin/env python3
# pylint: disable=invalid-name
"""BinarytoDecimal in Python"""

def binary_to_decimal(binary_number: int) -> int | str:
    """
    Convert a binary number to a decimal number.

    Return a string with an error if the binary
    number is not valid.
    """
    value = str(binary_number)
    length = len(value)
    output = 0
    for position in range(length):
        char = value[position]
        if char == "1":
            output += 2 ** (length - 1 - position)
        elif char == "0":
            continue
        else:
            return "Make sure you only input binary values"
    return output


if __name__ == "__main__":
    print(binary_to_decimal(int(input("Binary Number: "))))

File: BinarytoDecimal.hs
import System.IO

if_check :: (Char, Int, Int) -> Int
if_check ('0', len, pos) = 0
if_check ('1', len, pos) = (2 ^ (len - pos - 1))
if_check (char, len, pos) = error "Make sure you only input binary values"

binary_to_decimal :: Int -> Int
binary_to_decimal binary_number =
    sum [
        if_check (
            (show binary_number) !! position,
            length (show binary_number),
            position
        ) |
        position <- [0..length (show binary_number) - 1]
    ]

main :: IO ()
main = do
    putStr "Binary Number: "
    hFlush stdout
    input <- getLine
    putStrLn (show (binary_to_decimal (read input :: Int)))